linode/networkreservedips/framework_datasource.go

```diff
@@ -0,0 +1,120 @@
+package networkreservedips
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/linode/linodego"
+	"github.com/linode/terraform-provider-linode/v2/linode/helper"
+)
+
+func NewDataSource() datasource.DataSource {
+	return &DataSource{
+		BaseDataSource: helper.NewBaseDataSource(
+			helper.BaseDataSourceConfig{
+				Name:   "linode_reserved_ip",
+				Schema: &frameworkDataSourceSchema,
+			},
+		),
+	}
+}
+
+type DataSource struct {
+	helper.BaseDataSource
+}
+
+type DataSourceModel struct {
+	ID          types.String `tfsdk:"id"`
+	Address     types.String `tfsdk:"address"`
+	Region      types.String `tfsdk:"region"`
+	Gateway     types.String `tfsdk:"gateway"`
+	SubnetMask  types.String `tfsdk:"subnet_mask"`
+	Prefix      types.Int64  `tfsdk:"prefix"`
+	Type        types.String `tfsdk:"type"`
+	Public      types.Bool   `tfsdk:"public"`
+	RDNS        types.String `tfsdk:"rdns"`
+	LinodeID    types.Int64  `tfsdk:"linode_id"`
+	Reserved    types.Bool   `tfsdk:"reserved"`
+	ReservedIPs types.List   `tfsdk:"reserved_ips"`
+}
+
+func (data *DataSourceModel) parseIP(ip *linodego.InstanceIP) {
+	data.ID = types.StringValue(ip.Address)
+	data.Address = types.StringValue(ip.Address)
+	data.Region = types.StringValue(ip.Region)
+	data.Gateway = types.StringValue(ip.Gateway)
+	data.SubnetMask = types.StringValue(ip.SubnetMask)
+	data.Prefix = types.Int64Value(int64(ip.Prefix))
+	data.Type = types.StringValue(string(ip.Type))
+	data.Public = types.BoolValue(ip.Public)
+	data.RDNS = types.StringValue(ip.RDNS)
+	data.LinodeID = types.Int64Value(int64(ip.LinodeID))
+	data.Reserved = types.BoolValue(ip.Reserved)
+}
```

Comment:
Can you move this data source model and parsing function to `framwork_models` to keep the data source short and clear?

Reply:
Yeah I will be doing it the next commit